src/rgw/driver/rados/rgw_sync.cc

```diff
@@ -1942,6 +2079,28 @@ class RGWMetaSyncCR : public RGWCoroutine {
 
   int operate(const DoutPrefixProvider *dpp) override {
     reenter(this) {
+      yield {
+        //create watch object for sync lock bids
+        const auto& control_pool = sync_env->store->svc()->zone->get_zone_params().control_pool;
+        auto control_obj = rgw_raw_obj{control_pool, meta_sync_bids_oid};
+
+        auto bid_manager = rgw::sync_fairness::create_rados_bid_manager(
+            sync_env->store, control_obj, sync_status.sync_info.num_shards);
+        ret = bid_manager->start();
+        if (ret < 0) {
+          tn->log(0, SSTR("ERROR: failed to start bidding manager " << ret));
+          return ret;
+        }
+
+        //cr for broadcasting sync lock notifications in the background
+        sync_env->bid_manager = bid_manager.get();
+        call(new RGWMetaSyncShardNotifyCR(sync_env, tn));
+        if (retcode < 0) {
+          tn->log(5, SSTR("ERROR: failed to notify bidding information" << retcode));
+          continue;
+        }
```

Comment:
this notify coroutine should run in the background. by using `call()` here, `RGWMetaSyncCR` would block forever waiting for it to complete, and never go on to spawn the `ShardCR`s below

we'll need to avoid blocking on `RGWMetaSyncShardNotifyCR` by using `spawn()` instead of `call()`

with this extra background coroutine running, we'll also need to account for that in the logic below:
```diff
-        while (ret == 0 && num_spawned() > 0) {
+        while (ret == 0 && num_spawned() > 1) {
           yield wait_for_child();
           collect(&ret, nullptr);
         }
-        drain_all();
+        drain_all_but_stack(notify_stack);
```
at the very end before returning, you'll need to stop that notify cr (using something like `lease_cr->go_down()` or maybe `RGWCoroutinesStack::cancel()`?) and wait for it to drain too

this pattern shows up in other coroutines that have to account for the `ContinuousLeaseCR` running in the background

Reply:
oh right. call() creates cr on the same stack. thanks!

Reply:
trying `RGWCoroutinesStack::cancel()` here. the only usage seems to be in `RGWCoroutinesManager::run()`